### PR TITLE
chore: add coveralls

### PR DESCRIPTION
**Pull request by joshua-goldstein**
## Problem

We are not reporting test coverage.

## Solution

We show test coverage with coveralls.

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -4,16 +4,25 @@ set -eo pipefail
...
 # export packages because the test will run in a sub process.
 export packages=$(go list ./... | grep "github.com/dgraph-io/badger/v3/")
 
 tags="-tags=jemalloc"
 
-# Ensure that we can compile the binary.
+# Compile the Badger binary
 pushd badger
-go build -v $tags .
+  go build -v $tags .
```
Comment (skrdgraph) [inline test.sh:25]: fix the extra space* (I think it was added accidentally)

File: test.sh [hunk 2]
```diff
@@ -61,32 +82,44 @@ root() {
 
   echo "==> Running root level tests."
   set -e
-  go test $tags -timeout=25m . -v -race -parallel 16
+  go test $tags -v -race -parallel=16 -timeout=25m $covermode $coverprofile .
+  write_coverage
```
Comment (skrdgraph) [inline test.sh:86]: you could do this in 1line, with an `&&`, I think that would look cleaner

File: test.sh [hunk 3]
```diff
@@ -61,32 +82,44 @@ root() {
...
+    ./badger benchmark read --dir=$baseDir/test --full-scan | tee --append $baseDir/log.txt
+    ./badger benchmark read --dir=$baseDir/test -d=30s | tee --append $baseDir/log.txt
+    ./badger stream --dir=$baseDir/test -o "$baseDir/test2" | tee --append $baseDir/log.txt
+    count=$(cat "$baseDir/log.txt" | grep "at program end: 0 B" | wc -l)
+    rm -rf $baseDir
+    if [ $count -ne 4 ]; then
+      echo "LEAK detected in Badger stream."
+      return 1
+    fi
+    echo "==> DONE stream test"
```
Comment (skrdgraph) [inline test.sh:104]: just curious, why the extra space?

File: test.sh [hunk 4]
```diff
@@ -61,32 +82,44 @@ root() {
...
+      cat cover_tmp.out >> cover.out && rm cover_tmp.out
+    fi
+  fi
+  
+}
+
 export -f stream
 export -f manual
 export -f root
+export -f write_coverage
```
Comment (skrdgraph) [inline test.sh:122]: I think this is redundant, you may not need this line

File: test.sh (hunk 2 above)
Comment (joshua-goldstein) [reply test.sh:86]: Done 👍

File: test.sh (hunk 3 above)
Comment (joshua-goldstein) [reply test.sh:104]: I think that got pushed by accident, I reverted it

File: test.sh (hunk 4 above)
Comment (joshua-goldstein) [reply test.sh:109]: In that case do we need any of these export statements?  I'm not sure why we needed them to the environment

Comment (joshua-goldstein) [reply test.sh]: @skrdgraph  I removed all export statements, couldn't understand why they were there

**Review by skrdgraph — APPROVED** [review]

File: test.sh (hunk 4 above)
Comment (skrdgraph) [reply test.sh]: 👍

**Pull request merged**